packages/button/src/Button.tsx

```diff
@@ -1,483 +1,182 @@
+    }
+  }, [rippleRef, variant, darkMode]);
+
+  const isIconOnlyButton = ((leftGlyph || rightGlyph) && !children) ?? false;
+
+  const buttonClassName = getClassName({
+    variant,
+    size,
+    darkMode,
+    baseFontSize,
+    disabled,
+    showFocus,
+  });
+
+  const isAnchor = typeof rest.href === 'string';
+  let type: JSX.IntrinsicElements['button']['type'];
+
+  // @ts-expect-error rest.as may be defined
+  if ((rest.as && rest.as === 'button') || (!isAnchor && !rest.as)) {
+    type = 'button';
   }
-`;
-
-const disabledStyle = css`
-  color: ${uiColors.gray.base};
-  border-color: ${uiColors.gray.light1};
-  background-color: ${uiColors.gray.light2};
-  background-image: none;
-  box-shadow: none;
-  pointer-events: none;
-`;
 
-interface BaseButtonProps {
-  disabled?: boolean;
-  darkMode?: boolean;
-  variant?: Variant;
-  size?: Size;
-  glyph?: React.ReactElement;
-  className?: string;
-  borderRadius?: string;
-  children?: React.ReactNode;
-  href?: string;
-  forceState?: {
-    focused?: boolean;
-    active?: boolean;
-    // Supporting "hovered" requires factoring out all the &:hover
-    // styles from the the base style and each variant's styles
+  const buttonProps = {
+    type,
+    className: cx(buttonClassName, className),
+    ref: forwardRef,
+    // Provide a default value for the as prop
+    // If consumping application passes a value for as, it will override the default set here
+    as: isAnchor ? 'a' : 'button',
+    // only add a disabled prop if not an anchor
+    ...(typeof rest.href !== 'string' && { disabled }),
+    'aria-disabled': disabled,
+    ...rest,
+  } as const;
+
+  const accessibleIconProps = !isIconOnlyButton && {
+    'aria-hidden': true,
+    role: 'presentation',
   };
-}
-
-const Button: ExtendableBox<
-  BaseButtonProps & { ref?: React.Ref<any> },
-  'button'
-> = React.forwardRef(
-  (
-    {
-      className,
-      borderRadius,
-      children,
-      disabled = false,
-      darkMode = false,
-      variant = Variant.Default,
-      size = Size.Normal,
-      glyph,
-      forceState = {},
-      ...rest
-    }: BaseButtonProps,
-    ref: React.Ref<any>,
-  ) => {
-    const { usingKeyboard: showFocus } = useUsingKeyboardContext();
 
-    const commonProps = {
-      ref,
+  const clonedLeftGlyph =
+    leftGlyph &&
+    React.cloneElement(leftGlyph, {
       className: cx(
-        baseStyle,
-        buttonSizes[size],
-        buttonVariants[variant],
-        { [disabledStyle]: disabled },
-        { [focusStyle]: showFocus },
         {
           [css`
-            border-radius: ${borderRadius};
-
-            &:before,
-            &:after {
-              border-radius: calc(${borderRadius} - 1px);
-            }
-          `]: borderRadius !== undefined,
-        },
-        {
-          [css`
-            &:before {
-              opacity: 1;
-            }
-          `]: forceState.focused === true,
+            margin-right: ${iconSpacing[size]}px;
```

Comment:
I think that I prefer coupling the css call with the iconSpacing enum versus using an inline css call here.

Reply:
it's easier to do this inline because we use the same spacing values when we need to apply margin-right and margin-left. this allows us to not have to duplicate the iconSpacing const

Reply:
You can probably cheat a little here with `:first/last-of-type` selectors so you don't need to duplicate the const.

Reply:
i can't because what if only one is present? then it'll technically be the first of type even if it's on the right side

Reply:
```
&:first-of-type:last-of-type {
  margin-left: 0;
  margin-right: 0;
}
```

or

```
&:first-of-type:not(:last-of-type) {
  margin-right: 8px; // Or whatever it is
}
```

Reply:
👍